core/storage/trie/polkadot_trie/polkadot_trie_cursor.cpp

```diff
@@ -214,4 +239,17 @@ namespace kagome::storage::trie {
     last_visited_child_.emplace_back(std::make_pair(parent, child_idx));
   }
 
+  auto PolkadotTrieCursor::constructLastVisitedChildPath(
+      const common::Buffer &key)
+      -> outcome::result<
+          std::list<std::pair<PolkadotTrieCursor::BranchPtr, int8_t>>> {
+    OUTCOME_TRY(path, trie_.getPath(trie_.getRoot(), codec_.keyToNibbles(key)));
+    std::list<std::pair<PolkadotTrieCursor::BranchPtr, int8_t>>
+        last_visited_child;
+    for (auto &&[branch, idx] : path) {
+      last_visited_child.push_back({branch, idx});
```

Comment:
you could use emplace_back

Reply:
Should. =) Probably with std::move (depends on args' types).

Reply:
To be honest I don't see much difference between `this->_M_insert(end(), std::move(__x));` and `this->_M_insert(end(), std::forward<_Args>(__args)...);` for an rvalue of std::pair I pass there.

Reply:
But it looks nicer now that I replaced pair with a custom structure, so okay = )